src/cmdline/mod.rs

```diff
@@ -234,24 +218,117 @@ impl Cmdline {
     /// ```rust
     /// # use linux_loader::cmdline::*;
     /// # use std::ffi::CString;
-    /// let mut cl = Cmdline::new(100);
-    /// cl.insert_str("foobar");
-    /// let cl_cstring = CString::new(cl).unwrap();
-    /// assert_eq!(cl_cstring.to_str().unwrap(), "foobar");
+    /// let mut cl = Cmdline::new(100).unwrap();
+    /// cl.insert_str("foobar").unwrap();
+    /// assert_eq!(cl.as_cstring().unwrap().as_bytes_with_nul(), b"foobar\0");
     /// ```
     pub fn insert_str<T: AsRef<str>>(&mut self, slug: T) -> Result<()> {
-        let s = slug.as_ref();
+        let s = slug.as_ref().trim();
         valid_str(s)?;
 
-        self.has_capacity(s.len())?;
+        let mut cmdline_size = self.get_null_terminated_representation_size();
+
+        // Checking if one extra space needs to be added before insertion of a new boot arg
+        cmdline_size =
+            match cmdline_size.checked_add(if !self.boot_args.is_empty() { 1 } else { 0 }) {
+                None => {
+                    return Err(Error::TooLarge);
+                }
+                Some(new_cmdline_size) => new_cmdline_size,
+            };
+
+        // Adding extra size required for the insertion of the new boot args
+        cmdline_size = match cmdline_size.checked_add(s.len()) {
```

Comment:
We should convert Options into Results for neat code:
```rust
cmdline_size = cmdline_size.checked_add(s.len()).ok_or(Error::TooLarge)?;
```

Reply:
done